File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
  const map = key => {
    switch (key) {
      case 'type': return 0
      case 'color': return 1
      case 'name': return 2
      default: 3
    }
  }
  const key = map(ruleKey)
  return items.filter(a => a[key] === ruleValue).length
};

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
  ans := 0
  for _, v := range items {
    switch ruleKey {
    case "type": if v[0] == ruleValue { ans++ }
    case "color": if v[1] == ruleValue { ans++ }
    case "name": if v[2] == ruleValue { ans++ }
    }
  }
  return ans
}